manager/logbroker/broker.go

```diff
@@ -288,23 +335,47 @@ func (lb *LogBroker) PublishLogs(stream api.LogBroker_PublishLogsServer) error {
 		return err
 	}
 
+	var currentSubscription *subscription
+
+	// Make *sure* to call `done()` before returning from this function,
+	// otherwise the clients will be stuck on waiting for the stream to
+	// terminate.
+	done := func(err error) error {
```

Comment:
dang ... that helps triggering a race condition

When the publisher finishes and we hit `Done(this node)`, it triggers immediately the end of the stream on the other side.

However, there's no guarantee that we finished draining the publish queue.

Any thoughts on how to get around this?